.github/copilot-instructions.md

```diff
@@ -0,0 +1,268 @@
+# Copilot Instructions for esp-stub-lib
+
+## Repository Overview
+
+**esp-stub-lib** is an experimental C library for creating flash stubs that can be loaded onto Espressif ESP chips. The repository is ~5MB in size and contains a three-layer architecture designed to maximize code reuse across 13 different ESP chip targets while eliminating circular dependencies.
```

Comment:
We will continuously add more chips in the future. So I don't think we should be specific about the number.